src/transmem.cpp

```diff
@@ -200,23 +444,43 @@ bool TransMem::shortestPath(Path &path) const {
 
    }
 
-    // no path found
+    // No path found.
     return false;
-
 }
 
-void TransMem::dumpAsJSON() const {
+// Protected functions for debugging
 
-    QString path = "";
-    QJsonObject transmemObject;
+void TransMem::writeJSON(QJsonObject &json) const {
 
-    std::lock_guard<std::recursive_mutex> guard(lock);
+    QJsonArray frameObjects;
+    for(auto f: frameID2Frame){
+        QJsonObject frameObject;
+        f.second.writeJSON(frameObject);
+        frameObjects.append(frameObject);
+    }
 
-    writeJSON(transmemObject);
+    QJsonArray linkObjects;
+    for(Link l: links){
```

Comment:
You might want to use auto for l (and name it link), for the sake of consistency with the code above.